classes/test.php

```diff
@@ -1242,6 +1242,11 @@ public function runTestMethod($testMethod, array $tags = array())
 						}
 					);
 
+					$this->assertionManager->setPropertyHandler('newMockInstance', function($class, $mockNamespace = null, $mockClass = null, array $constructorArguments = array()) {
+							return $this->mockGenerator->getNewMockInstance($class, $mockNamespace, $mockClass, $constructorArguments);
```

Comment:
Here i would use a `new`like that: 

```php
$namespace = $this->mockGenerator->getDefaultNamespace();
$classname = rtrim($namespace, '\\') . '\\' . ltrim($class, '\\');

return new $classname;

```

This will cover most use-cases. If the user provides some constructor arguments we might rely on reflection to instanciate the mock but this should not be the default behavior.